### PR TITLE
Do not discard data in case track set change

### PR DESCRIPTION
**Pull request by youennf**
Replace 'immediately stop' by 'stop' as immediately does not seem to add anything

### CODE REVIEW

File: MediaRecorder.bs [hunk 1]
```diff
@@ -223,8 +223,8 @@ interface MediaRecorder : EventTarget {
       </li>
 
       <li>If at any point, a track is added to or removed from the {{MediaRecorder/stream}}'s
-      track set, the UA MUST immediately stop gathering data, discard any data that it has gathered,
-      and queue a task, using the DOM manipulation task source, that runs the following steps:
+      track set, the UA MUST stop gathering data, and queue a task,
```
Comment (yell0wd0g) [inline MediaRecorder.bs]: What happened to "discard any data that it has gathered," ?

File: MediaRecorder.bs [hunk 2]
```diff
@@ -209,7 +209,7 @@ interface MediaRecorder : EventTarget {
 
       <li>If at any point the {{MediaRecorder/stream}}'s <a>isolation properties</a>
       change so that {{MediaRecorder}} is no longer allowed access to it, the
-      UA MUST immediately stop gathering data, discard any data that it has
+      UA MUST stop gathering data, discard any data that it has
```
Comment (yell0wd0g) [inline MediaRecorder.bs]: The changes do not correspond with the title of the PR: 
  "Do not discard data in case track set change"

Comment (Pehrsons) [reply MediaRecorder.bs]: This one is due to a change in the stream's isolation properties, not its track set.
Some backstory on "immediately" in [issue 151 ](https://github.com/w3c/mediacapture-record/issues/151#issuecomment-457446229).

File: MediaRecorder.bs (hunk 1 above)
Comment (Pehrsons) [reply MediaRecorder.bs]: See [this comment](https://github.com/w3c/mediacapture-record/issues/151#issuecomment-456756089) on issue 151. Seems to me like a copy/paste error when the track set change algorithm was defined.

File: MediaRecorder.bs (hunk 2 above)
Comment (guest271314) [reply MediaRecorder.bs]: @youennf Note, the specification does not explicitly prohibit _replacing_ the track, see 

> [4.8.12.10 Media resources with multiple media tracks](https://html.spec.whatwg.org/multipage/media.html#media-resources-with-multiple-media-tracks)
> 
> A media resource can have multiple embedded audio and video tracks. For example, in addition to the primary video and audio tracks, a media resource could have foreign-language dubbed dialogues, director's commentaries, audio descriptions, alternative angles, or sign-language overlays.
> 
> There are only ever one AudioTrackList object and one VideoTrackList object per media element, even if another media resource is loaded into the element: the objects are reused. (The AudioTrack and VideoTrack objects are not, though.)

https://github.com/w3c/mediacapture-record/issues/147#issuecomment-480580164

Comment (guest271314) [reply MediaRecorder.bs]: @Pehrsons Related https://bugzilla.mozilla.org/show_bug.cgi?id=1557394

**Comment by Pehrsons** [comment]
@yellowdoge is there anything blocking the merge of this PR?

**Comment by Pehrsons** [comment]
@jan-ivar perhaps you can nudge this along?

**Review by jan-ivar — APPROVED** [review]

**Pull request merged**